### PR TITLE
[vstest/subintf] Add vs test to validate sub interface ingress to a vnet

### PR DESCRIPTION
**Pull request by wendani**
**What I did**
Extended the following vs tests to validate the use case of sub interface ingress linked to a vnet:
- sub port interface creation
- sub port interface add ipv4 addresses
- sub port interface admin status change
- sub port interface remove ipv4 addresses
- sub port interface removal

**Why I did it**
Add vs test to validate that https://github.com/Azure/sonic-swss/pull/1521 fixes https://github.com/Azure/sonic-buildimage/issues/4238

**How I verified it**
vs tests

**Details if related**

vnet is a special vrf with the following limitations as of the current implementation:

- For next hop object of type IP, only directly connected subnet and single next hop to an IP prefix are supported
- Directly connected subnet to a router interface is limited to IPv4 when interface is ingress linked to a vnet

vs tests are extended to vnet with the above limitations in mind.

#### Dedendancy
- [x] Contains https://github.com/Azure/sonic-swss/pull/1521
- [x] Contains https://github.com/Azure/sonic-swss/pull/1654

Incremental change to https://github.com/Azure/sonic-swss/pull/1521 base can be found at https://github.com/wendani/sonic-swss/pull/1

### CODE REVIEW

File: tests/conftest.py [hunk 1]
```diff
@@ -979,16 +979,19 @@ def set_interface_status(self, interface, admin_status):
         tbl.set(interface, fvs)
         time.sleep(1)
 
-    # deps: acl, fdb_update, fdb, mirror_port_erspan, vlan
-    def add_ip_address(self, interface, ip):
+    # deps: acl, fdb_update, fdb, mirror_port_erspan, vlan, sub port intf
+    def add_ip_address(self, interface, ip, vrf_name=""):
```
Comment (lguohan) [inline tests/conftest.py:983]: use None for the default value?

**Comment by lguohan** [comment]
it is unclear which pr to review this one or #1521 , can you advise?

**Comment by wendani** [comment]
> it is unclear which pr to review this one or #1521 , can you advise?

vs test for vnet is recently developed. The intention of creating of a separate PR while #1521 is still pending is to limit the size of change per PR to facilitate review work. It is preferred to review and accept #1521 first, followed by this one.

File: orchagent/intfsorch.cpp [hunk 2]
```diff
@@ -731,6 +731,7 @@ void IntfsOrch::doTask(Consumer &consumer)
                     {
                         IntfsEntry intfs_entry;
                         intfs_entry.ref_count = 0;
+                        intfs_entry.proxy_arp = false;
```
Comment (prsunny) [inline orchagent/intfsorch.cpp:734]: Is this required? For `lo `interfaces, we are not continuing with the set flow. So just curious why initialize here?

File: cfgmgr/intfmgr.cpp [hunk 3]
```diff
@@ -428,16 +428,15 @@ bool IntfMgr::doIntfGeneralTask(const vector<string>& keys,
 
     string alias(keys[0]);
     string vlanId;
-    string subIntfAlias;
+    string parentAlias;
```
Comment (prsunny) [inline cfgmgr/intfmgr.cpp:431]: Why this change? Its not matching the PR description

**Review by prsunny — COMMENTED** [review]
lgtm.. @vsenchyshyn, @neethajohn for viz

File: cfgmgr/intfmgr.cpp (hunk 3 above)
Comment (wendani) [reply cfgmgr/intfmgr.cpp:431]: It is the change in https://github.com/Azure/sonic-swss/pull/1521. This PR only contains change in vs test, developed on top of https://github.com/Azure/sonic-swss/pull/1521. However, without https://github.com/Azure/sonic-swss/pull/1521, vs test will fail

File: orchagent/intfsorch.cpp (hunk 2 above)
Comment (wendani) [reply orchagent/intfsorch.cpp:734]: Just for object data member init completeness. I think it does no harm.

**Comment by neethajohn** [comment]
@wendani, can you please rebase on top of the latest master since #1521 is already in master

**Comment by wendani** [comment]
> @wendani, can you please rebase on top of the latest master since #1521 is already in master

Not yet in master. Can you help merge #1521?

**Comment by neethajohn** [comment]
> > @wendani, can you please rebase on top of the latest master since #1521 is already in master
> 
> Not yet in master. Can you help merging #1521?

Done!

**Comment by neethajohn** [comment]
@prsunny , can you please review?

**Review by shi-su — APPROVED** [review]
